solver/src/driver.rs

```diff
@@ -414,6 +421,14 @@ impl Driver {
 
             tracing::debug!("winning settlement: {:?}", settlement);
             self.submit_settlement(settlement.clone()).await;
+            self.inflight_orders = settlement
+                .settlement
+                .settlement
+                .trades()
+                .iter()
+                .map(|t| (t.order.order_meta_data.uid, t.executed_amount))
+                .collect::<HashMap<OrderUid, U256>>();
```

Comment:
Why not use your fancy type alias?
```suggestion
                .collect::<InflightOrders>();
```

Also, I suspect the compiler would be able to figure this out on its own:
```suggestion
                .collect();
```

Reply:
Actually it is going to be eliminated in favour of @fleupold's suggestion above. However in the future it might reappear as

`InflightTrades - HashMap<OrderUid, Trade>`

Reply:
I added comments :point_up: